### PR TITLE
[DataGrid] Add Sudanese Arabic (arSD) locale

### PR DESCRIPTION
**Pull request by YassinHussein**
add Arabic Localization

### CODE REVIEW

**Comment by flaviendelangle** [comment]
Thanks for your help !

https://app.circleci.com/pipelines/github/mui-org/material-ui-x/6515/workflows/b13f8c6d-c8d4-41a4-8bd4-980e11ef19f9/jobs/34895

You need to run `yarn prettier` to reformat your code

https://app.circleci.com/pipelines/github/mui-org/material-ui-x/6515/workflows/b13f8c6d-c8d4-41a4-8bd4-980e11ef19f9/jobs/34891

The core do not have the `arSD` locale. 
@m4theushw do we agree that we need the core translations in order to add the grid one
Or do we accept to use the english for the core ?

File: packages/grid/_modules_/grid/locales/arSD.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+  // Checkbox selection text
+  checkboxSelectionHeaderName: 'تحديد',
+
+  // Boolean cell text
+  booleanCellTrueLabel: 'صحيح',
+  booleanCellFalseLabel: 'خاطئة',
+
+  // Used core components translation keys
+  MuiTablePagination: {},
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/locales/arSD.ts:107]:
```suggestion
```

File: packages/grid/_modules_/grid/locales/arSD.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+};
+
+export const arSD: Localization = getGridLocalization(arSDGrid, arSDCore);
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/locales/arSD.ts:110]: This locale doesn't exist in the [core](https://material-ui.com/guides/localization/#supported-locales) yet. Do you want to open another PR there to add the translations for the pagination or is it fine to use the English as default?

```suggestion
export const arSD: Localization = getGridLocalization(arSDGrid, enUSCore);
```

File: packages/grid/_modules_/grid/locales/arSD.ts (hunk 1 above)
Comment (YassinHussein) [reply packages/grid/_modules_/grid/locales/arSD.ts]: updated

File: packages/grid/_modules_/grid/locales/arSD.ts (hunk 2 above)
Comment (YassinHussein) [reply packages/grid/_modules_/grid/locales/arSD.ts]: updated

**Comment by oliviertassinari** [comment]
The change in the core won't be accessible to us until we drop v4 support, looking at #27588, we did a plain copy and paste of the arEG locale… so we could do the same here

**Comment by m4theushw** [comment]
> The change in the core won't be accessible to us until we drop v4 support. Looking at [mui-org/material-ui#27588](https://github.com/mui-org/material-ui/pull/27588), we did a plain copy and paste of the arEG locale… so we could do the same here

@oliviertassinari My idea would be to extend the core arEG or enUS locale, but use arSD if available.

File: packages/grid/_modules_/grid/locales/arSD.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
+import { enUS as enUSCore } from '@material-ui/core/locale';
```
Comment (oliviertassinari) [inline packages/grid/_modules_/grid/locales/arSD.ts:1]:
```suggestion
// TODO v5: import from arSD
import { enUS as enUSCore } from '@material-ui/core/locale';
```

**Review by oliviertassinari — APPROVED** [review]
> My idea would be to extend the core arEG or enUS locale, but use arSD if available.

@m4theushw This logic would be great, but AFAIK it's not possible to implement. We still need to retrain the tree-shaking property.
